### ISSUE TITLE
etcd shows as unhealthy in cs with bad certificate error

### ISSUE THREAD

**Issue by marcoceppi** [juju-solutions/bundle-canonical-kubernetes#261]
Not sure why, as the cluster is healthy

```
ubuntu@is-charming:~$ kubectl get cs
NAME                 STATUS      MESSAGE                                                                    ERROR
controller-manager   Healthy     ok                                                                         
scheduler            Healthy     ok                                                                         
etcd-1               Unhealthy   Get https://172.31.31.68:2379/health: remote error: tls: bad certificate   
etcd-0               Unhealthy   Get https://172.31.1.175:2379/health: remote error: tls: bad certificate   
etcd-2               Unhealthy   Get https://172.31.9.171:2379/health: remote error: tls: bad certificate   
ubuntu@is-charming:~$ kubectl cluster-info
Kubernetes master is running at https://54.215.162.71:443
Heapster is running at https://54.215.162.71:443/api/v1/proxy/namespaces/kube-system/services/heapster
KubeDNS is running at https://54.215.162.71:443/api/v1/proxy/namespaces/kube-system/services/kube-dns
kubernetes-dashboard is running at https://54.215.162.71:443/api/v1/proxy/namespaces/kube-system/services/kubernetes-dashboard
Grafana is running at https://54.215.162.71:443/api/v1/proxy/namespaces/kube-system/services/monitoring-grafana
InfluxDB is running at https://54.215.162.71:443/api/v1/proxy/namespaces/kube-system/services/monitoring-influxdb

To further debug and diagnose cluster problems, use 'kubectl cluster-info dump'.
ubuntu@is-charming:~$ juju ssh etcd/0
ubuntu@ip-172-31-31-68:~$ /snap/bin/etcd.etcdctl cluster-health
member a394e95b9a4fa76d is healthy: got healthy result from https://172.31.9.171:2379
member b304996ba1d4c62f is healthy: got healthy result from https://172.31.31.68:2379
member ed16262bc1aea9b2 is healthy: got healthy result from https://172.31.1.175:2379
cluster is healthy
```

**Comment by chuckbutler** [comment]
Hi marco, thanks for filing this bug. There's some background information in the upstream project issue tracker linked below:

https://github.com/kubernetes/kubernetes/issues/29330

and 

https://github.com/kubernetes/kubernetes/pull/39716


This 'tls error' shown is with regard to the kubernetes components not respecting the TLS certificates supplied. We have enabled strict TLS key verification on etcd to ensure there is a security policy in place. In relation to this, the k8s components not using those TLS keys are causing a false positive to be reported in the component status view.

Once the upstream project has resolved the TLS key issue, the CS display bug should be fixed. Until such a time there's very little we can do to resolve the CS output showing TLS errors, as we're not willing to risk security for a large number of deployments for green status.

**Comment by chuckbutler** [comment]
A fix is incoming with 1.7 which landed in https://github.com/kubernetes/kubernetes/pull/39716

**Comment by marcoceppi** [comment]
This has been fixed for a while.
